interface-definitions/nat_cgnat.xml.in

```diff
@@ -0,0 +1,143 @@
+                </properties>
+                <children>
+                  <leafNode name="external-port-range">
+                    <properties>
+                      <help>Port range</help>
+                      <valueHelp>
+                        <format>range</format>
+                        <description>Numbered port range (e.g., 1001-1005)</description>
+                      </valueHelp>
+                      <constraint>
+                        <validator name="port-range"/>
+                      </constraint>
+                    </properties>
+                    <defaultValue>1024-65535</defaultValue>
+                  </leafNode>
+                  <node name="per-user-limit">
+                    <properties>
+                      <help>Per user limits for the pool</help>
+                    </properties>
+                    <children>
+                      <leafNode name="port">
+                        <properties>
+                          <help>Ports per user</help>
+                          <valueHelp>
+                            <format>u32:1-65535</format>
+                            <description>Numeric IP port</description>
+                          </valueHelp>
+                          <constraint>
+                            <validator name="numeric" argument="--range 1-65535"/>
+                          </constraint>
+                        </properties>
+                        <defaultValue>2000</defaultValue>
+                      </leafNode>
+                    </children>
+                  </node>
+                  <tagNode name="range">
+                    <properties>
+                      <help>Range of IP addresses</help>
+                      <valueHelp>
+                        <format>ipv4net</format>
+                        <description>IPv4 prefix</description>
+                      </valueHelp>
+                      <valueHelp>
+                        <format>ipv4range</format>
+                        <description>IPv4 address range</description>
+                      </valueHelp>
+                      <constraint>
+                        <validator name="ipv4-prefix"/>
+                        <validator name="ipv4-host"/>
+                        <validator name="ipv4-range"/>
+                      </constraint>
+                    </properties>
+                    <children>
+                      <leafNode name="seq">
+                        <properties>
+                          <help>Sequence</help>
+                        </properties>
+                      </leafNode>
+                    </children>
+                  </tagNode>
+                </children>
+              </tagNode>
+              <tagNode name="internal">
+                <properties>
+                  <help>Internal pool settings</help>
+                </properties>
+                <children>
+                  <leafNode name="range">
```

Comment:
Can we combine external and internal range to one XML include block?

Reply:
Probably not
The external pool should use sequences for the range to add priority to external address allocation.
The internal pool does not have sequences. However, we can add sequences to the internal pool if we want.
```
set nat cgnat pool external ext1 range 192.0.2.222/32 seq 1
set nat cgnat pool external ext1 range 192.0.2.253/32 seq 2
```
But internal is a single pool
```
set nat cgnat pool internal int1 range '100.64.0.0/28'
```